### PR TITLE
CPS-83: Added LoggerAdapter.

### PR DESCRIPTION
**Pull request by JaviCerveraIngram**
In order to correctly show log messages in Azure Application Insights, with their request & asset ids, the way a message is formatted needs to change.

Previously, ids were directly prepended to format string in the logger formatter. This makes this information not being sent to Azure, since it only shows the %(message) field. Now, a LoggerAdapter is being used instead, which embeds the ids in the %(message) itself, so it is shown in Azure.

### CODE REVIEW

**Comment by JaviCerveraIngram** [comment]
Closing PR by now since I am considering extending the use of LoggerAdapters to Automation classes custom loggers, which would be a bit cleaner than the current approach. Will give it a thought and reopen.

File: connect/resources/automation_engine.py [hunk 1]
```diff
@@ -39,34 +42,40 @@ def process_request(self, request):
...
+        if not self._logger_adapter or self._logger_adapter.logger.name != name:
+            self._logger_adapter = LoggerAdapter(logging.getLogger(name))
+        return self._logger_adapter
+
+    def _set_current_request(self, request):
+        # type: (Optional[BaseModel]) -> None
+        self._current_request = request
+        self._set_logger_prefix(request)
+
+    def _set_logger_prefix(self, request):
```
Comment (JaviCerveraIngram) [inline connect/resources/automation_engine.py:79]: This method will be overridden by the subclasses, providing the correct request information for each type of request.

File: connect/logger/logger.py [hunk 2]
```diff
@@ -17,27 +17,29 @@
...
+        return (
+            '[%s] %s' % (self.prefix, msg) if self.prefix else msg,
+            kwargs
+        )
+
+    def setLevel(self, level):
+        self.logger.setLevel(level)
+
+
+def function_log(func):
```
Comment (JaviCerveraIngram) [inline connect/logger/logger.py:35]: Before, function_log was accepting the custom_logger used by each Automation class. This has been simplified and is no longer neccessary.

File: connect/resources/automation_engine.py [hunk 3]
```diff
@@ -16,19 +15,23 @@
...
 
-    @function_log(custom_logger=logger)
+    @function_log
     def process(self, filters=None):
-        '''
-        # type: (Dict[str, Any]) -> None
-        '''
+        # # type: (Dict[str, Any]) -> None
         for request in self.list(filters):
+            self._set_current_request(request)
```
Comment (JaviCerveraIngram) [inline connect/resources/automation_engine.py:32]: For each request, the request to process is stored so the correct prefix is used in the LoggerAdapter of the AutomationEngine instance. This approach works fine with multithreading.

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**